code/modules/mob/living/carbon/xenomorph/castes/runner/abilities_runner.dm

```diff
@@ -194,3 +194,90 @@
+
+
+/datum/action/xeno_action/evasion/action_activate()
+	var/mob/living/carbon/xenomorph/runner/R = owner
+
+	R.do_jitter_animation(2000)
+	R.visible_message("<span class='warning'>[R.name] begins to move erratically!</span>", \
+	"<span class='xenodanger'>We move erratically, making us impossible to hit with projectiles; the next [RUNNER_EVASION_STACKS] projectile damage that would hit us will now miss.</span>")
+
+	addtimer(CALLBACK(src, .proc/evasion_warning), RUNNER_EVASION_DURATION * RUNNER_EVASION_DURATION_WARNING) //Warn the runner when the duration is about to expire.
+	addtimer(CALLBACK(src, .proc/evasion_deactivate), RUNNER_EVASION_DURATION)
+
+	R.evasion_stacks = RUNNER_EVASION_STACKS
+
+	succeed_activate()
+
+	GLOB.round_statistics.runner_evasions++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "runner_evasions") //Statistics
+	add_cooldown()
+
+
+/datum/action/xeno_action/evasion/proc/evasion_deactivate()
+
+	var/mob/living/carbon/xenomorph/runner/R = owner
+	if(R.evasion_stacks) //If our evasion stacks are already depleted, don't tell us again.
+		R.evasion_stacks = 0
+		R.visible_message("<span class='warning'>[R.name] stops moving erratically.</span>", \
+		"<span class='highdanger'>We stop moving erratically; projectiles will hit us normally again!</span>")
+
+	R.playsound_local(R, 'sound/voice/hiss5.ogg', 50)
+
+
+/datum/action/xeno_action/evasion/on_cooldown_finish()
+	to_chat(owner, "<span class='xenodanger'>We are able to take evasive action again.</span>")
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+/datum/action/xeno_action/evasion/proc/evasion_warning()
+
+	to_chat(owner,"<span class='highdanger'>We begin to slow down as we tire. We can only keep this up for [RUNNER_EVASION_DURATION * (1-RUNNER_EVASION_DURATION_WARNING) * 0.1] more seconds!</span>")
+	owner.playsound_local(owner, 'sound/voice/hiss4.ogg', 50, 0, 1)
+
+
+/mob/living/carbon/xenomorph/runner/projectile_hit(obj/projectile/proj, cardinal_move, uncrossing)
+
+	if(evasion_stacks && (last_move_intent > (world.time - RUNNER_EVASION_RUN_DELAY) ) ) //Gotta keep moving to benefit from evasion!
+
+		do_jitter_animation(4000) //Dodgy animation!
+
+		if(issamexenohive(proj.firer)) //We automatically dodge allied projectiles at no cost
+			return FALSE
+
+		evasion_stacks -= (proj.damage)
+
+		visible_message("<span class='warning'>[name] effortlessly dodge the [proj.name]!</span>", \
+		"<span class='xenodanger'>We effortlessly dodge the [proj.name]![evasion_stacks > 0 ? " We can dodge [evasion_stacks] more projectile damage." : ""]</span>")
+
+		var/turf/T = get_turf(src) //after image SFX
+		playsound(T, pick('sound/effects/throw.ogg','sound/effects/alien_tail_swipe1.ogg', 'sound/effects/alien_tail_swipe2.ogg'), 25, 1) //sound effects
+		var/i = 0
+		var/obj/effect/temp_visual/xenomorph/runner_afterimage/A
+		while(i < rand(2,4)) //random number of after images
```

Comment:
please make it a fixed amount of 'after images' 1 for preference. They should still dodge each individual shotgun pellet.